### DISCUSSION TITLE
consult-completion-in-region, lsp-mode method signatures

### DISCUSSION THREAD

**Discussion by stevemolitor** [minad/consult#1047] [Q&A]
Using corfu + marginalia in lsp-mode I can see method signatures and documentation for completion candidates:

![](https://cdn.zappy.app/1c9390bd882fc047931f4e95d3e1d167.png)

With `consult-completion-in-region` I only see the method names:

![](https://cdn.zappy.app/069daafd7ef0dc60030f128157e5e54e.png)

Is it possible to get at least the method signatures when using `consult-completion-in-region` in lsp-mode? 

I don't use auto-completion. I only use on-demand completion.

EDIT: Alternately is there a way to use embark to show the signature or documentation on demand for a candidate within the minibuffer? That might be ideal actually.

**Comment by minad** [comment]
Unfortunately I cannot give you much assistance regarding `lsp-mode` and `consult-completion-in-region`, since I don't use `lsp-mode` and `consult-completion-in-region` only sparingly. As mentioned in the [README](https://github.com/minad/consult?tab=readme-ov-file#miscellaneous), `consult-completion-in-region` is not expected to work well with `lsp-mode`, since the `lsp-completion-at-point-function` expects the input to arrive in the original buffer (and not in the minibuffer). That being said, I cannot explain why annotations don't work in this case. In principle they should work. Maybe the `lsp-mode` maintainers have some insight?
